nx_parallel/algorithms/shortest_paths/dense.py

```diff
@@ -0,0 +1,250 @@
+        The default blocking factor is get by finding the optimal value
+        for the core available
+
+    Returns
+    -------
+    A : 2D array
+        All pairs shortest paths Graph adjacency matrix
+
+    References
+    ----------
+    .. [1] Gary J. Katz and Joseph T. Kider Jr:
+    All-Pairs Shortest-Paths for Large Graphs on the GPU, 2008.
+
+    """
+    if hasattr(G, "graph_object"):
+        G = G.graph_object
+    undirected = not G.is_directed()
+    nodelist = list(G)
+    A = _adjacency_matrix(G, weight, nodelist, undirected)
+    n = G.number_of_nodes()
+
+    total_cores = nxp.get_n_jobs()
+
+    if blocking_factor is None:
+        blocking_factor, is_prime = _find_nearest_divisor(n, total_cores)
+    no_of_primary = n // blocking_factor
+
+    for primary_block in range(no_of_primary):
+        k_start = (primary_block * n) // no_of_primary
+        k_end = k_start + (n // no_of_primary) - 1
+        if is_prime and primary_block == no_of_primary - 1:
+            k_end = k_end + (n % no_of_primary)
+        k = (k_start, k_end)
+        # Phase 1: Compute Primary block
+        # Execute Normal floyd warshall for the primary block submatrix
+        _partial_floyd_warshall_numpy(A, k, k, k)
+
+        # Phase 2: Compute Cross block
+
+        params = []
+        for block in range(no_of_primary):
+            # skip the primary block computed in phase 1
+            if block != primary_block:
+                # append the actual indices of the matrix by multiply the block number with the blocking factor
+                if is_prime and block == no_of_primary - 1:
+                    block_coord = (block * blocking_factor, n - 1)
+                else:
+                    block_coord = _block_range(blocking_factor, block)
+                params.append((block_coord, k))
+                params.append((k, block_coord))
+
+        Parallel(n_jobs=(no_of_primary - 1) * 2, require="sharedmem")(
+            delayed(_partial_floyd_warshall_numpy)(A, k, i, j) for (i, j) in params
+        )
+
+        # Phase 3: Compute remaining
+        params.clear()
+        for block_i in range(no_of_primary):
+            for block_j in range(no_of_primary):
+                # skip all block previously computed, so skip every block with primary block value
+                if block_i != primary_block or block_j != primary_block:
+                    i_range = _block_range(blocking_factor, block_i)
+                    j_range = _block_range(blocking_factor, block_j)
+                    if is_prime:
+                        if block_i == no_of_primary - 1:
+                            i_range = (block_i * blocking_factor, n - 1)
+                        if block_j == no_of_primary - 1:
+                            j_range = (block_j * blocking_factor, n - 1)
+                    params.append((i_range, j_range))
+        Parallel(n_jobs=(no_of_primary - 1) ** 2, require="sharedmem")(
```

Comment:
Recently we have made it so that all the parameters of the `Parallel` call are controlled by the user and we don't add anything in it. I would highly encourage you(@Fre0Grella) to join our weekly nx-parallel meetings :  https://scientific-python.org/calendars/networkx.ics , if you can. I think your insights and feedback would really help us shape the nx-parallel project in a better way. Thank you :)